internal/controller/rhdh/backstage.go

```diff
@@ -36,121 +34,166 @@ var ConfigMapNameAndConfigDataKey = map[string]string{
+	if err != nil {
+		knativeLogger.Error(err, "Error occurred when checking subscription exists", "SubscriptionName", BackstageSubscriptionName)
+		return err
+	}
+	if !subscriptionExists {
+		if err := kubeoperations.InstallOperatorViaSubscription(ctx, client, olmClientSet, BackstageOperatorGroup, rhdhSubscription); err != nil {
+			knativeLogger.Error(err, "Error occurred when installing operator", "SubscriptionName", BackstageSubscriptionName)
+			return err
+		}
+		knativeLogger.Info("Operator successfully installed", "SubscriptionName", BackstageSubscriptionName)
+	}
+
+	if subscriptionExists {
+		// Compare the current and desired state
+		if !reflect.DeepEqual(existingSubscription.Spec, rhdhSubscription.Spec) {
+			// Set owner reference for proper garbage collection
+			//if err := controllerutil.SetControllerReference(&orchestrator, oslSubscription, r.Scheme); err != nil {
+			//	return err
+			//}
+
+			// Update the existing subscription with the new Spec
+			existingSubscription.Spec = rhdhSubscription.Spec
+			if err := client.Update(ctx, existingSubscription); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func CreateBSSecret(secretNamespace string, ctx context.Context, client client.Client) error {
 	logger := log.FromContext(ctx)
 	logger.Info("Creating Backstage NPMrc Secret")
 
 	secret := &corev1.Secret{}
 	err := client.Get(ctx, types.NamespacedName{
 		Namespace: secretNamespace,
-		Name:      secretName,
+		Name:      RegistrySecretName,
 	}, secret)
 
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			logger.Info("Secret does not exist. Creating secret", "Secret", secretName)
+			logger.Info("Secret does not exist. Creating secret", "Secret", RegistrySecretName)
 			newSecret := &corev1.Secret{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      secretName,
+					Name:      RegistrySecretName,
 					Namespace: secretNamespace,
-					Labels:    operations.AddLabel(),
+					Labels:    kubeoperations.AddLabel(),
 				},
 				Type: corev1.SecretTypeOpaque,
 				StringData: map[string]string{
-					".npmrc": fmt.Sprintf("registry=%s", npmRegistry),
+					".npmrc": fmt.Sprintf("registry=%s", NpmRegistry),
 				},
 			}
 
 			if err := client.Create(ctx, newSecret); err != nil {
-				logger.Error(err, "Error occurred when creating secret", "Secret", secretName)
+				logger.Error(err, "Error occurred when creating secret", "Secret", RegistrySecretName)
 				return err
 			}
-			logger.Info("Successfully created secret", "Secret", secretName)
+			logger.Info("Successfully created secret", "Secret", RegistrySecretName)
 			return nil
 		}
-		logger.Error(err, "Error occurred when checking secret exist", "Secret", secretName)
+		logger.Error(err, "Error occurred when checking secret exist", "Secret", RegistrySecretName)
 		return err
 	}
-	logger.Info("Secret already exist", "Secret", secretName)
+	logger.Info("Secret already exist", "Secret", RegistrySecretName)
 	return nil
 }
 
 func HandleCRCreation(
```

Comment:
a CR represents an instance or something else? I find the naming to generic, `HandleInstanceCreation` may be clearer as to what is happening in the function, because from what I understand, here we are create a new backstage/RHDH CR that will be digested and processed by the RHDH operator

Reply:
Changed it to `HandleRHDHCR` mostly because there will be an update to the CR and will extend this func to handle that scenario.